icefoganalytics/travel-authorization
# Part 1: Wire Travel Desk Request Tab

Relates to:

- [TravelDeskWireFrame.pdf](https://github.com/icefoganalytics/travel-authorization/files/14016343/TravelDeskWireFrame.pdf)
- [UserStories-TravelDesk.docx](https://github.com/icefoganalytics/travel-authorization/files/14016344/UserStories-TravelDesk.docx)
- [TravelDeskDataModel (1).xlsx](https://github.com/icefoganalytics/travel-authorization/files/14016345/TravelDeskDataModel.1.xlsx)
- https://github.com/ytgov/travel-authorization/tree/main/web/src/modules/travelDesk
- Part of solving https://github.com/icefoganalytics/travel-authorization/issues/167

# Context

Using the existing Travel Desk forms (https://github.com/ytgov/travel-authorization/tree/main/web/src/modules/travelDesk) rewire the travel desk request to exist in the travel desk tab of the travel auth.

Travel Desk forms should be created anytime a user is traveling by air.

![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/74adc941-50cd-4219-a723-062822c4a94a)

# Implementation

Add base travel desk request into my-travel-requests "Request" tab. Identify that the components don't work outside of there current location and begin migrating said components to something that will work. Do this by cloning any component tree that is broken into a new more traditional model based namespace. Avoid making any changes to UI unless absolutely necessary. Focus on un-breaking code, but lint and simplify as necessary.
Focus on reducing component complexity and having components only do one thing. i.e. create/update but not both.

Add user sync failure concept to avoid trying to sync user that fail sync. This speeds up development app loading by 30+ seconds after first failure.

## Current Progress

Create the travel desk request as part of the "ApproveService" if the travel plan includes travel by air.
Travel Details saves correctly, though lacks validation for Internation travel and contact information.

# Screenshots

Current partially-working version of Travel Desk Request in Request tab
![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/ea3187db-6d2e-4042-a653-76096dd0fcf1)

# Testing Instructions

1. Run the test suite via `dev test` (or `dev test_api`)
2. Boot the app via `dev up`
3. Log in to the app at http://localhost:8080
4. Go to the "My Travel Requests" page via the top drop down nav.
5. Create a travel authorization.
6. In the details section, select at least one travel method with "aircraft" type.
7. Generate the travel estimate.
8. Submit the travel authorization to yourself as the supervisor.
9. Go to Manger View page accessible via the drop down nav.
10. Click on the new request from the "Pending Approvals" panel in the top left.
11. Scroll to the bottom and approve the request.
12. Go back to the "My Travel Requests" page, and find your new travel request in the table.
13. Click on the Submit Travel Desk Request button to go to the Request tab. (NEW: it works now!)
14. Check that the Traveler Details load correctly, and that you can save them.
15. Check that when you enable international travel, the international travel fields become required.
16. Check that when you enable "contact information different for travel" the travel contact fields become required.
